test/python/scheduler/test_basic_scheduler.py

```diff
@@ -614,3 +614,13 @@ def test_multiple_measure_in_3Q(self):
             (10, acquire(AcquireChannel(2), MemorySlot(1)))
         )
         self.assertEqual(sched.instructions, expected.instructions)
+
+    def test_alap_keeps_circuit_same(self):
+        """Test that scheduling does not change the input circuit."""
+        q = QuantumRegister(2)
+        c = ClassicalRegister(2)
+        qc = QuantumCircuit(q, c)
+        qc.cx(q[0], q[1])
+        qc_sched = qc + QuantumCircuit(*qc.qregs)
```

Comment:
here it's better to explicitly copy to make sure we are really comparing to the original (unmodified) circuit
```suggestion
        qc_copy = qc.copy()
```